File: backend/models/Payment.js
import mongoose from 'mongoose';

const paymentSchema = new mongoose.Schema(
  {
    staff: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    periodStart: { type: Date, required: true },
    periodEnd: { type: Date, required: true },
    amount: { type: Number, required: true },
    status: { type: String, enum: ['pending', 'paid'], default: 'pending' },
    paymentMethod: { type: String, enum: ['cash', 'razorpay'], default: null },
    razorpayPaymentId: { type: String, default: null },
    razorpayOrderId: { type: String, default: null },
    razorpaySignature: { type: String, default: null },
  },
  { timestamps: true }
);

export default mongoose.model('Payment', paymentSchema);

File: my_app/lib/src/core/models/payment.dart
enum PaymentStatus { pending, paid }

class WorkerPayment {
  final String id;
  final String workerId;
  final DateTime periodStart;
  final DateTime periodEnd;
  final double amount;
  final PaymentStatus status;
  final String? paymentMethod; // 'cash', 'razorpay', etc.
  final String? razorpayPaymentId;
  final String? razorpayOrderId;

  const WorkerPayment({
    required this.id,
    required this.workerId,
    required this.periodStart,
    required this.periodEnd,
    required this.amount,
    required this.status,
    this.paymentMethod,
    this.razorpayPaymentId,
    this.razorpayOrderId,
  });
}


